Warn about third-party JVM agents in docs (#99718)
elastic/elasticsearch: docs/reference/setup/install.asciidoc

takes responsibility for keeping it up to date. Security issues and bugs within
the bundled JVM are treated as if they were within {es} itself.

> You should not use long-running third-party agents that attach to the JVM

Reviewer: Remove "long-running"? Any java agent, regardless of how long lived, should not be attached to Elasticsearch.

for deep introspection and instrumentation. Such software has been observed

> to reduce Elasticsearch performance, or even freeze or crash nodes.

Reviewer 2: I think we should have fewer qualifiers here. It's not just _long-running agents for instrumentation_ that we want to exclude, running with any kind of agent attached is unsupported.

Also I'd prefer stronger wording than "you should not" - ideally "do not" or something equally imperative.

The bundled JVM is located within the `jdk` subdirectory of the {es} home
directory. You may remove this directory if using your own JVM.

[discrete]
[[jvm-agents]]

> === JVM and Java Agents

Reviewer 3: ```suggestion
=== JVM and Java agents
```

> Do not use third-party Java agents that attach to the JVM. Such software has

Reviewer 3: ```suggestion
Don't use third-party Java agents that attach to the JVM. These agents
```

> been observed to reduce {es} performance, or even freeze or crash nodes.

Reviewer 3: ```suggestion
can reduce {es} performance, including freezing or crashing nodes.
```
